tests/stdlib/thttpclient_standalone.nim

```diff
@@ -5,25 +5,42 @@ discard """
 import asynchttpserver, httpclient, asyncdispatch, strutils
 
 block: # bug #16436
-  proc startServer() {.async.} =
+  proc startServer(): AsyncHttpServer =
+    result = newAsyncHttpServer()
+    result.listen(Port(0))
+
+  proc runServer(server: AsyncHttpServer) {.async.} =
+    var killServer = false
+
     proc cb(req: Request) {.async.} =
       let headers = { "Content-length": "15"} # Provide invalid content-length
+      killServer = true
       await req.respond(Http200, "Hello World", headers.newHttpHeaders())
 
-    var server = newAsyncHttpServer()
-    await server.serve(Port(5555), cb)
+    while not killServer:
+      if server.shouldAcceptRequest():
+        await server.acceptRequest(cb)
+      else:
+        poll()
 
-  proc runClient() {.async.} =
+  proc runClient(port: Port) {.async.} =
     let c = newAsyncHttpClient(headers = {"Connection": "close"}.newHttpHeaders)
-    let r = await c.getContent("http://127.0.0.1:5555")
+    discard await c.getContent("http://127.0.0.1:" & $(uint16)port)
```

Comment:
What is this `$(uint16)port` syntax? It kinda looks like a C cast...

Reply:
it's the same as `$uint16(port)`, a consequence of command invocation syntax. `f x` is valid and then you put the `f` in parenthesis, `(f) x`

Reply:
I see, why not just use `$uint16(port)` then? That is much clearer imo.

Reply:
Agreed.

Reply:
Sorry, I never know which form to use; `uint16(port)`, `port.uint16`, `uint16 port`, or `(uint16) port`.  In C it would be a cast so I made it look like a cast.  I've changed it to the suggested form.  If a particular form is unacceptable, it should probably be marked as deprecated and eventually removed.

Reply:
for cast, it's cast[Foo](bar), but we shouldn't use cast here

> marked as deprecated and eventually removed.

never encountered this form `(foo)bar` before your PR, so maybe it's not a big problem in practice; it's just regular application of parser rules. Although one could make a case that `(foo)bar` serves no purpose compared to `foo bar` when foo is a symbol, and could give a deprecation warning for readability's sake.

Reply:
Understood.  I don't know where I picked up (foo)bar from then.  I've possibly just gotten confused with another language.  I'll avoid that form in future.